modules/apigee_edge_teams/src/Entity/ListBuilder/TeamListBuilder.php

```diff
@@ -110,4 +110,33 @@ public function buildRow(EntityInterface $entity) {
     return $row + parent::buildRow($entity);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function render() {
+    $build = parent::render();
+    $account = $this->entityTypeManager->getStorage('user')->load(\Drupal::currentUser()->id());
+
+    // Team lists vary for each user.
+    $build['#cache']['contexts'][] = 'user';
+
+    // This team list should be invalidated if the user's permissions change.
+    $build['#cache']['tags'][] = 'user:' . $account->id();
+
+    // This team list should be invalidated if the user roles permissions change.
+    foreach ($account->getRoles() as $rid) {
```

Comment:
Isn't it covered by the user:$id tag or the user user context already? What happens if a user gets a new role and that should cause the invalidation of this page? The new role is of course not available in the list of the _current_ roles of the user.

Reply:
Yep, it turned out this was covered by the cache context.